### DISCUSSION TITLE
How to prevent SWR revalidating if key changed in SWR hook

### DISCUSSION THREAD

**Discussion by phatify** [vercel/swr#2236] [Q&A]
I have issue when using **useSWR/useSWRInfinite** hook to fetch data. Specifically, there are 3 variables in key of swr hook that I get and it changes whenever.

For example:

```
const usePost = (profile_id, from_id = '') => {
    const PAGE_LIMIT = 5;
    let { data, size, setSize, error, mutate, isValidating } = useSWRInfinite(
        (pageIndex, previousPageData) => {
            const currentPageIndex = pageIndex + 1;
            return `/posts/profile/${profile_id}?page=${currentPageIndex}&from_id=${from_id}`;
        },
        fetcher,
        {
            revalidateFirstPage: false,
        },
    );
    // more code
}
```

How can I **prevent** useSWRInfinite/useSWR revalidating even if from_id changes (excludes **profile_id** and **currentPageIndex**) ?

**Comment by koba04** [comment]
In this case, the key of the `useSWRInfite` is `currentPageIndex` and `profile_id`, so you return the values as the key and combine them in the fetcher function.

```jsx
const usePost = (profile_id, from_id = '') => {
    const PAGE_LIMIT = 5;
    let { data, size, setSize, error, mutate, isValidating } = useSWRInfinite(
        (pageIndex, previousPageData) => {
            const currentPageIndex = pageIndex + 1;
            return [currentPageIndex, profile_id];
        },
        ([currentPageIndex, profile_id]) => {
            return fetcher(`/posts/profile/${profile_id}?page=${currentPageIndex}&from_id=${from_id}`)
        },
        {
            revalidateFirstPage: false,
        },
    );
    // more code
}
```

**Reply by phatify** [reply]
@koba04  thanks, it works perfectly. But in this case, I want to use my custom function `setFromId()` instead of `setSize()` and then **the key of the useSWRInfite**  will return a callback function with args is **fromId**.
e.g:  
```
const { data, setFromId } = useSWRInfinite(
    (fromId) => {
        return fromId;
    },
   (fromId) => {
        return fetcher(`/posts/profile/${profile_id}?from_id=${fromId}`)
   }
)

//...
setFromId ('abcd1234');
```

This is all i expected, because pagination may not be completely **pageIndex**. Is there any solution for this problem?
